crates/ruff/src/settings/types.rs

```diff
@@ -244,3 +282,35 @@ impl Deref for Version {
         &self.0
     }
 }
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct RequiresVersion(pub Vec<VersionSpecifier>);
+
+impl<'de> Deserialize<'de> for RequiresVersion {
```

Comment:
This is great, i've stolen this to integrate it into pep440-rs itself: https://github.com/konstin/pep440-rs/commit/0d15f41b5b5c76c58278575b815d5cfe0bb54787

Reply:
Thanks, I pulled the new version.